packages/mds-ingest-service/repository/index.ts

```diff
@@ -199,14 +219,61 @@ class IngestReadWriteRepository extends ReadWriteRepository {
         query.andWhere('events.provider_id = ANY(:provider_ids)', { provider_ids })
       }
 
-      const entities = await query.limit(limit).getMany()
+      // Use query instead of paginator to manage order if using a joined field
+      if (order && order.column === 'vehicle_id') {
+        query.orderBy('devices.vehicle_id', order.direction)
```

Comment:
I'm unsure how the `query.orderBy` and `paginationKeys` interact, but I assume the pagination keys will get appended to the orderBy. Is `timestamp` always the secondary sort column?

Reply:
Good call out, I took a look at this and it turns out that `paginationKeys` always override anything added to the query with `orderBy` or `addOrderBy`. If you remove `paginationKeys` param entirely, it just uses `id` (the default).

There seems to be an open issue for this (can't sort on relational tables / override `paginationKeys`) - and the maintainer has ack'd it recently.
https://github.com/benjamin658/typeorm-cursor-pagination/issues/4

So for now, removed the ability to sort by `vehicle_id`. Will follow up with the open issue. Not worth sinking a bunch of time for this one sort-able field atm.

That being said, sorting on joined tables will most likely come up in the future at some point - so hopefully this gets addressed in typeorm-cursor-pagination shortly.